frontend/packages/process-editor/src/components/ConfigPanel/ConfigEndEvent/CustomReceiptContent/CreateCustomReceiptForm/CreateCustomReceiptForm.tsx

```diff
@@ -0,0 +1,134 @@
+import { getExistingDatamodelIdFromLayoutsets } from '../../../../../utils/customReceiptUtils';
+import { getLayoutSetIdValidationErrorKey } from 'app-shared/utils/layoutSetsUtils';
+
+export type CreateCustomReceiptFormProps = {
+  onCloseForm: () => void;
+};
+
+export const CreateCustomReceiptForm = ({
+  onCloseForm,
+}: CreateCustomReceiptFormProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const { layoutSets, existingCustomReceiptLayoutSetId, addLayoutSet, mutateDataType } =
+    useBpmnApiContext();
+
+  const [layoutSetError, setLayoutSetError] = useState<string>(null);
+  const [datamodelError, setDatamodelError] = useState<string>(null);
+
+  const existingDatamodelId: string = getExistingDatamodelIdFromLayoutsets(
+    layoutSets,
+    existingCustomReceiptLayoutSetId,
+  );
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const formData: FormData = new FormData(event.currentTarget);
+
+    const customReceiptForm: CustomReceiptType = {
+      layoutSetId: formData.get('customReceiptLayoutSetId') as string,
+      datamodelId: formData.get('customReceiptDatamodel') as string,
+    };
+    const { layoutSetId, datamodelId } = customReceiptForm;
+
+    if (layoutSetId && datamodelId) {
+      saveCustomReceipt(customReceiptForm);
+    }
+
+    const validationResult = handleValidateLayoutSetId(layoutSetId);
+    setLayoutSetError(validationResult);
+    setDatamodelError(
+      !datamodelId
+        ? t('process_editor.configuration_panel_custom_receipt_create_datamodel_error')
+        : null,
+    );
+  };
+
+  const saveCustomReceipt = (customReceipt: CustomReceiptType) => {
+    const isSameLayoutId: boolean = existingCustomReceiptLayoutSetId === customReceipt.layoutSetId;
+    const isSameDatamodelId: boolean = existingDatamodelId === customReceipt.datamodelId;
+
+    if (isSameLayoutId && isSameDatamodelId) return;
+
+    if (!existingCustomReceiptLayoutSetId) {
+      createNewCustomReceipt(customReceipt);
+      return;
+    }
+  };
+
+  const createNewCustomReceipt = (customReceipt: CustomReceiptType) => {
+    const customReceiptLayoutSetConfig: LayoutSetConfig = {
+      id: customReceipt.layoutSetId,
+      tasks: [PROTECTED_TASK_NAME_CUSTOM_RECEIPT],
+    };
+    addLayoutSet(
+      {
+        layoutSetIdToUpdate: customReceipt.layoutSetId,
+        layoutSetConfig: customReceiptLayoutSetConfig,
+      },
+      {
+        onSuccess: () => saveDatamodel(customReceipt.datamodelId),
+      },
+    );
+  };
+
+  const saveDatamodel = (datamodelId: string) => {
+    const dataTypeChange: DataTypeChange = {
+      newDataType: datamodelId,
+      connectedTaskId: PROTECTED_TASK_NAME_CUSTOM_RECEIPT,
+    };
+    mutateDataType(dataTypeChange, {
+      onSuccess: onCloseForm,
+    });
+  };
+
+  const handleChangeLayoutsetName = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
It is done like this to avoid having the code in the onChange function. 
I think it looks messy when having it like this: 
```tsx
        onChange={(e: React.ChangeEvent>HTMLInputElement>) => {
             setLayoutSetError(handleValidateLayoutSetId(e.target.value));
        }
```

So, to make the code more readable and the JSX more clean, I moved it into its own function. 

However, I can move the setLayoutSetError function into the validation function to remove the duplicate usage of it 😄

Reply:
See my previous comment 😄